test/with_dummyserver/test_socketlevel.py

```diff
@@ -152,6 +152,24 @@ def socket_handler(listener):
 
         self.assertEqual(http.pool.qsize(), http.pool.maxsize)
 
+    def test_read_timeout_dont_retry_method_not_in_whitelist(self):
+        timed_out = Event()
+        def socket_handler(listener):
+            sock = listener.accept()[0]
+            while not sock.recv(65536):
```

Comment:
Hrm. This is a slightly odd construction. This loop will only execute once, and in the event the connection comes up and is then closed it'll loop a second time and then IOError. What were you aiming for here?